src/main/python/ayab/plugins/ayab_plugin/ayab_control.py

```diff
@@ -85,20 +91,20 @@ def get_progress(self):
 
     def __get_knit_func(self):
         '''Select function that decides which line of data to send according to the machine type and number of colors'''
-        if self.__knitting_mode == KnittingMode.SINGLEBED.value and self.__numColors == 2:
+        if self.__knitting_mode == KnittingMode.SINGLEBED.value and self._numColors == 2:
             self.__knit_func = self.__singlebed_2col
-        elif self.__knitting_mode == KnittingMode.CLASSIC_RIBBER_1.value and self.__numColors == 2:
-            self.__knit_func = self.__doublebed_2col
-        elif self.__knitting_mode == KnittingMode.CLASSIC_RIBBER_1.value and self.__numColors > 2:
-            self.__knit_func = self.__doublebed_multicol
-        elif self.__knitting_mode == KnittingMode.MIDDLECOLORSTWICE_RIBBER.value:
-            self.__knit_func = self.__middlecoltwice
-        elif self.__knitting_mode == KnittingMode.HEARTOFPLUTO_RIBBER.value and self.__numColors >= 2:
-            self.__knit_func = self.__heartofpluto
-        elif self.__knitting_mode == KnittingMode.CIRCULAR_RIBBER.value and self.__numColors == 2:
-            self.__knit_func = self.__circular_ribber
+        elif self.__knitting_mode == KnittingMode.CLASSIC_RIBBER_1.value and self._numColors == 2:
+            self.__knit_func = self._doublebed_2col
+        elif self.__knitting_mode == KnittingMode.CLASSIC_RIBBER_1.value and self._numColors > 2:
+            self.__knit_func = self._doublebed_multicol
+        elif self.__knitting_mode == KnittingMode.MIDDLECOLORSTWICE_RIBBER.value and self._numColors >= 2:
+            self.__knit_func = self._middlecoltwice
+        elif self.__knitting_mode == KnittingMode.HEARTOFPLUTO_RIBBER.value and self._numColors >= 2:
+            self.__knit_func = self._heartofpluto
+        elif self.__knitting_mode == KnittingMode.CIRCULAR_RIBBER.value: # not restricted to 2 colors
```

Comment:
I could not see any necessary reason to restrict circular knitting to 2 colours. Am I missing something?

Reply:
I have no idea ;) @Adrienne200 might be able to answer that question. Meanwhile, perhaps https://manual.ayab-knitting.com/how_to_knit/circular_knitting/ gives an answer.